### PR TITLE
Frontend Assignment 2

### PR DESCRIPTION
**Pull request by asish918**
## Drift Video
[Drift Video](https://video.drift.com/v/abniT2lxQtbhFj7moWkTwHVr5K4QUpAyQg2CWq9dAQMv0)

I have attached screenshots of a few things I forgot to show in my drift video.

## Login Screen Incase of Invalid Credentials
Screenshot 2024-02-18 at 22 34 37

## Initial store of posts and users (I am using the picsum/photos link for all of them)
Screenshot 2024-02-18 at 22 29 50
Screenshot 2024-02-18 at 22 29 35

## I am storing images as blob objects. Its a big dirty string
Screenshot 2024-02-18 at 22 28 08

## GET Request for fetching all posts
Screenshot 2024-02-18 at 22 27 51

## Homepage Desktop View
Screenshot 2024-02-18 at 22 02 53

## Homepage Mobile View
Screenshot 2024-02-18 at 22 03 30

## Active Users
Screenshot 2024-02-18 at 22 03 54

## Chats
Screenshot 2024-02-18 at 22 04 51

### CODE REVIEW

**Comment by asish918** [comment]
| Metric | Value |
| ------ | ----- |
| alert_status | <b>:white_check_mark: OK</b> |
| bugs | 2 |
| code_smells | 12 |
| reliability_rating | C (at least 1 Major Bug)<br/>Note: A being best, E being worst |
| security_rating | A (0 Vulnerabilities)<br/>Note: A being best, E being worst |
| vulnerabilities | 0 |

File: frontend/assignments/2/server.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+const PORT = process.env.PORT || 8000;
+const io = new Server(server);
+
+const { userJoin, getCurrentUser, userLeave, getUsers, getChats, addMessage } = require('./memory/users');
+
+const authApi = require('./routes/authApi');
+const userApi = require('./routes/userApi');
+const postApi = require('./routes/postApi');
+
+
```
Comment (Nitesh2503) [inline frontend/assignments/2/server.js:20]: Format the code properly

File: frontend/assignments/2/server.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+const path = require('path');
+const { Server } = require('socket.io');
+const morgan = require('morgan');
+
+const app = express();
+const server = http.createServer(app);
+const PORT = process.env.PORT || 8000;
+const io = new Server(server);
+
+const { userJoin, getCurrentUser, userLeave, getUsers, getChats, addMessage } = require('./memory/users');
```
Comment (Nitesh2503) [inline frontend/assignments/2/server.js:14]: Remove the unwanted imports

File: frontend/assignments/2/memory/users.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    messages.get(user[0].id).push({
+        content,
+        time: "25:08",
+        type: "self",
+        user_id: id
+    });
+
+    if (user[0].id !== id) {
+
```
Comment (Nitesh2503) [inline frontend/assignments/2/memory/users.js:55]: Format the code properly

File: frontend/assignments/2/public/sass/message.scss [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+  height: 100vh;
+  gap: 10px;
+  align-items: flex-start;
+  width: 23%;
+  margin-left: 5%;
+
+  &__post-button {
+    background-color: $primary-color;
+    width: 80%;
+    height: 50px;
```
Comment (Nitesh2503) [inline frontend/assignments/2/public/sass/message.scss:19]: Basically not needed

File: frontend/assignments/2/public/sass/message.scss [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+    color: $secondary-text-color;
+  }
+}
+
+.message {
+  width: 40%;
+  min-height: 60px;
+
+  &__container {
+    // overflow-y: scroll;
```
Comment (Nitesh2503) [inline frontend/assignments/2/public/sass/message.scss:133]: Don't push the formatted code

File: frontend/assignments/2/public/sass/message.scss [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+
+  &__time {
+    font-size: 0.8rem;
+    color: $secondary-text-color;
+  }
+
+  &__input {
+    display: flex;
+    align-items: center;
+    background-color: rgba(136, 136, 136, 0.501);
```
Comment (Nitesh2503) [inline frontend/assignments/2/public/sass/message.scss:154]: Store it in the constants

**Pull request closed without merging**